examples/templates/clusterclass.yaml

```diff
@@ -0,0 +1,259 @@
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3ClusterTemplate
+metadata:
+  name: example-cluster-template
+spec:
+  template:
+    spec:
+      controlPlaneEndpoint:
+        host: 127.0.0.1
+        port: 6443
+      noCloudProvider: true
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlaneTemplate
+metadata:
+  name: example-controlplane
+spec:
+  template:
+    spec:
+      kubeadmConfigSpec:
+        clusterConfiguration: {}
+        initConfiguration:
+          nodeRegistration:
+            name: '{{ ds.meta_data.local_hostname }}'
+        joinConfiguration:
+          controlPlane: {}
+          nodeRegistration:
+            name: '{{ ds.meta_data.local_hostname }}'
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3MachineTemplate
+metadata:
+  name: example-md-0
+spec:
+  nodeReuse: false
+  template:
+    spec:
+      automatedCleaningMode: metadata
+      image:
+        checksum: ${IMAGE_CHECKSUM}
+        checksumType: ${IMAGE_CHECKSUM_TYPE}
+        format: ${IMAGE_FORMAT}
+        url: ${IMAGE_URL}
+      dataTemplate:
+        name: example-md-metadata
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfigTemplate
+metadata:
+  name: example-md-0
+spec:
+  template:
+    spec:
+      joinConfiguration:
+        nodeRegistration:
+          name: '{{ ds.meta_data.hostname }}'
```

Comment:
This doesn't work for me(virtualized env). Shouldn't it be `'{{ ds.meta_data.local_hostname }}'` or `'{{ ds.meta_data.name }}'`?. The question is also, what is the difference between `local_hostname` and `name` because I see both here.

Reply:
We can fix it in a followup or leave it out completely from the template in a followup PR. @p-strusiewiczsurmacki-mobica would you be interested in a followup PR? For now, we take this PR in

Reply:
@kashifest I'll try to take a look at this as soon as I can. However if anyone wants to work on this before I'll be able to, then go ahead. :)